compiler/test/fail_compilation/diag9679.d

```diff
@@ -1,13 +1,23 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/diag9679.d(11): Error: variable `diag9679.main.n` - only parameters, functions and `foreach` declarations can be `ref`
-fail_compilation/diag9679.d(12): Error: variable `diag9679.main.n` - storage class `auto` has no effect if type is not inferred, did you mean `scope`?
+fail_compilation/diag9679.d(14): Error: variable `diag9679.main.n` - storage class `auto` has no effect if type is not inferred, did you mean `scope`?
+fail_compilation/diag9679.d(15): Error: variable `diag9679.main.S.a` - field declarations cannot be `ref`
+fail_compilation/diag9679.d(22): Error: returning `r` escapes a reference to local variable `r`
```

Comment:
`r` escapes reference to `r`? 
`r` is a reference to `i`, not to itself.

Reply:
fixed it